tests/ui/cognitive_complexity.stderr

```diff
@@ -240,29 +216,45 @@ LL | | }
    |
    = help: you could split it up into multiple smaller functions
 
-error: the function has a cognitive complexity of 1
-  --> $DIR/cognitive_complexity.rs:340:1
+error: the function has a cognitive complexity of 13
+  --> $DIR/cognitive_complexity.rs:344:1
    |
-LL | / fn early() -> Result<i32, &'static str> {
-LL | |     return Ok(5);
-LL | |     return Ok(5);
-LL | |     return Ok(5);
+LL | / fn early_ret() -> i32 {
+LL | |     let a = if true { 42 } else { return 0; };
+LL | |     let a = if a < 99 { 42 } else { return 0; };
+LL | |     let a = if a < 99 { 42 } else { return 0; };
 ...  |
-LL | |     return Ok(5);
+LL | |     }
 LL | | }
    | |_^
    |
    = help: you could split it up into multiple smaller functions
 
-error: the function has a cognitive complexity of 8
-  --> $DIR/cognitive_complexity.rs:354:1
+error: the function has a cognitive complexity of 2
+  --> $DIR/cognitive_complexity.rs:364:1
    |
-LL | / fn early_ret() -> i32 {
-LL | |     let a = if true { 42 } else { return 0; };
-LL | |     let a = if a < 99 { 42 } else { return 0; };
-LL | |     let a = if a < 99 { 42 } else { return 0; };
-...  |
-LL | |     }
+LL | / fn osscilating_logical_chain_1() -> bool {
+LL | |     true && false || true && false
+LL | | }
+   | |_^
+   |
+   = help: you could split it up into multiple smaller functions
+
+error: the function has a cognitive complexity of 2
+  --> $DIR/cognitive_complexity.rs:371:1
+   |
+LL | / fn osscilating_logical_chain_2() -> bool {
+LL | |     true && false && true && false || true && false && true && false
+LL | | }
+   | |_^
+   |
+   = help: you could split it up into multiple smaller functions
+
+error: the function has a cognitive complexity of 2
+  --> $DIR/cognitive_complexity.rs:376:1
+   |
+LL | / fn osscilating_logical_chain_3() -> bool {
+LL | |     (true && false) || (true && false)
```

Comment:
Not something blocking or anything, but this stderr file is getting quite long and it would be nice to split it into multiple files by some criteria. cc #2038

Reply:
Good idea. Will do that, possibly tomorrow ^^
There are many different topics by which you can split the testing of this lint, so a split actually fits quite well.

Reply:
Maybe also create a subdir for testing this lint. I can imagine, that we will need many tests for this.

Reply:
I've been doing this for a while now. It's not done yet, but it certainly made things much cleaner! :)

I've been splitting them up by type:

* Nesting-Dependent Structures
* Nesting-Independent Structures
* No-Ops (non-scoring cases)
* Too Complex By Default (functions that trigger the default threshold)
* Other Cases (miscellaneous patterns)

Reply:
That said, the tests for the Binary Logical Operators Chain are just a ton :sweat_smile: . I might end up splitting the NI-Structures test and make one test just for testing that part of the lint.